### PR TITLE
feat(starfish): span sample list sidebar highlighting

### PR DESCRIPTION
**Pull request by DominikB2014**
This PR makes a bunch of changes to the span sample list, and some other supporting changes as well.
1. Adds highlighting between the duration chart and span sample table (video of this behaviour in slack)
2. Adds ability to click into a sample event from the graph
3. Change table column to Compared to baseline
4. Right align `at baseline` in the compared to baseline column
5. Right align `span summary` table header

There is one bug that I'm aware of, when you try to hover on the graph, and there's two events at the exact same time, only one will be highlighted. This is something i'll fix in another PR!

### CODE REVIEW

File: static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx [hunk 1]
```diff
@@ -26,36 +27,54 @@ const COLUMN_ORDER: TableColumnHeader[] = [
...
   };
-  'transaction.id': string;
-};
+} & SpanSample;
 
 type Props = {
   data: SpanTableRow[];
   isLoading: boolean;
   p95: number;
+  highlightSpanId?: string;
```
Comment (gggritso) [inline static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx:49]: I think this should be called `highlightedSpanId` or something similar, with an `ed` suffix. `highlightSpanId` implies it's some kind of function, or handler, or something!

File: static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx [hunk 2]
```diff
@@ -67,42 +86,56 @@ export function SpanSamplesTable({isLoading, data, p95}: Props) {
   }
 
   function renderBodyCell(column: GridColumnHeader, row: SpanTableRow): React.ReactNode {
```
Comment (gggritso) [inline static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx:88]: Passing these `commonProps` to every cell is definitely a bummer! I think `GridEditable` should support passing hover handlers to the grid row. Maybe something to bring up in the channel, or TSC meeting on Thursday?

File: static/app/views/starfish/views/spanSummaryPage/sampleList/durationChart/index.tsx [hunk 3]
```diff
@@ -88,17 +104,56 @@ function DurationChart({groupId, transactionName, transactionMethod}: Props) {
...
+    if (isSpanSample && onClickSample) {
+      const [timestamp, duration] = e.value as [string, number];
+      const sample = getSample(timestamp, duration);
+      if (sample) {
+        onClickSample(sample);
+      }
+    }
+  };
+
+  const hanldeChartHighlight: EChartHighlightHandler = e => {
```
Comment (gggritso) [inline static/app/views/starfish/views/spanSummaryPage/sampleList/durationChart/index.tsx:127]:
```suggestion
  const handleChartHighlight: EChartHighlightHandler = e => {
```
typo!

File: static/app/views/starfish/views/spanSummaryPage/sampleList/index.tsx [hunk 4]
```diff
@@ -14,6 +15,7 @@ type Props = {
 
 export function SampleList({groupId, transactionName, transactionMethod}: Props) {
   const router = useRouter();
+  const [highlightedSpanId, highlightSample] = useState<string | undefined>(undefined);
```
Comment (gggritso) [inline static/app/views/starfish/views/spanSummaryPage/sampleList/index.tsx:18]: This name mismatch is a little weird!
```suggestion
  const [highlightedSpanId, setHighlightedSpanId] = useState<string | undefined>(undefined);
```

would make more sense IMO

File: static/app/views/starfish/views/spanSummaryPage/sampleList/index.tsx [hunk 5]
```diff
@@ -37,12 +39,23 @@ export function SampleList({groupId, transactionName, transactionMethod}: Props)
...
         transactionName={transactionName}
         transactionMethod={transactionMethod}
+        onClickSample={span => {
+          router.push(
+            `/performance/${span.project}:${span['transaction.id']}/#span-${span.span_id}`
+          );
+        }}
+        onMouseOverSample={sample => highlightSample(sample.span_id)}
+        onMouseLeaveSample={() => highlightSample(undefined)}
+        highlightSpanId={highlightedSpanId}
```
Comment (gggritso) [inline static/app/views/starfish/views/spanSummaryPage/sampleList/index.tsx:49]: Similar mismatch here, why not
```suggestion
        highlightedSpanId={highlightedSpanId}
```

**Review by gggritso — APPROVED** [review]
Makes sense overall! Some parts are a little quirky, but would need changes to `GridEditable` to work well. A bunch of nits, but nothing major 👍🏻

File: static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx (hunk 1 above)
Comment (DominikB2014) [reply static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx]: That's a good point, will make the change!

File: static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx (hunk 2 above)
Comment (DominikB2014) [reply static/app/views/starfish/components/samplesTable/spanSamplesTable.tsx:88]: Yes, 100% a bummer and I definitely would want to have some sort of row event handlers! I think i'll bring it up the the team/TSC and work on it post release. I wanted to do get this out quicker for the wen release.

**Pull request merged**